mocks/service_mock.go

```diff
@@ -38,10 +38,16 @@ func (m *ServiceMock) StartServer() error {
 	return nil
 }
 
+// Not used: backward compatibility with public interface
 func (m *ServiceMock) ShutdownServer() {
-	m.server.Shutdown(context.TODO())
+	_ = m.ShutdownServerContext(context.TODO())
+}
+
+func (m *ServiceMock) ShutdownServerContext(ctx context.Context) (err error) {
```

Comment:
Let's use unnamed return value, just like in all other methods around. Just to keep the coding style unified.